core/toolbox.js

```diff
@@ -588,6 +589,43 @@ Blockly.Toolbox.TreeNode.prototype.onDoubleClick_ = function(e) {
 };
 
 /**
+ * Decorate original onKeyDown handler to swap
+ * LEFT/RIGHT keys with UP/DOWN in horizontalLayout.
+ * @param {!goog.events.BrowserEvent} e The browser event.
+ * @return {boolean} The handled value.
+ * @override
+ * @private
+ */
+Blockly.Toolbox.TreeNode.prototype.onKeyDown = function(e) {
+  var ret;
+
+  if (this.horizontalLayout_) {
+    var right = goog.events.KeyCodes.RIGHT;
+    var left = goog.events.KeyCodes.LEFT;
+    var up = goog.events.KeyCodes.UP;
+    var down = goog.events.KeyCodes.DOWN;
+
+    // Swap keycodes
```

Comment:
Swapping keycodes in an interrupt seems like a bad idea.  How about changing the key code in the event instead?